wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -149,94 +150,133 @@ def __init__(self, hostname, protocol="https", port=8443, k_entry="api/v1", o_en
 
         self._connect()
 
     def _connect(self):
-        if self.new_client:
-            url = '{proto}://{host}:{port}'.format(proto=self.protocol, host=self.hostname,
-                                                   port=self.port)
-            ociclient.configuration.host = url
-            kubeclient.configuration.host = url
-
-            ociclient.configuration.verify_ssl = self.verify_ssl
-            kubeclient.configuration.verify_ssl = self.verify_ssl
-
-            kubeclient.configuration.debug = self.debug
-            ociclient.configuration.debug = self.debug
-
-            token = 'Bearer {token}'.format(token=self.token)
-            ociclient.configuration.api_key['authorization'] = token
-            kubeclient.configuration.api_key['authorization'] = token
-            self.ociclient = ociclient
-            self.kclient = kubeclient
-            self.o_api = ociclient.OapiApi()
-            self.k_api = kubeclient.CoreV1Api()
-
-    def list_route(self):
+        url = '{proto}://{host}:{port}'.format(proto=self.protocol, host=self.hostname,
+                                               port=self.port)
+        ociclient.configuration.host = url
+        kubeclient.configuration.host = url
+
+        ociclient.configuration.verify_ssl = self.verify_ssl
+        kubeclient.configuration.verify_ssl = self.verify_ssl
+
+        kubeclient.configuration.debug = self.debug
+        ociclient.configuration.debug = self.debug
+
+        token = 'Bearer {token}'.format(token=self.token)
+        ociclient.configuration.api_key['authorization'] = token
+        kubeclient.configuration.api_key['authorization'] = token
+        self.ociclient = ociclient
+        self.kclient = kubeclient
+        self.o_api = ociclient.OapiApi()
+        self.k_api = kubeclient.CoreV1Api()
+
+    def list_route(self, namespace=None):
         """Returns list of routes"""
-        entities = []
-        entities_j = self.o_api.get('route')[1]['items']
-        for entity_j in entities_j:
-            meta = entity_j['metadata']
-            entity = Route(self, meta['name'], meta['namespace'])
-            entities.append(entity)
-        return entities
-
-    def list_docker_registry(self):
-        """Returns list of docker registries"""
-        entities = []
-        entities_j = self.o_api.get('imagestream')[1]['items']
-        for entity_j in entities_j:
-            if 'dockerImageRepository' not in entity_j['status']:
-                continue
-            meta = entity_j['metadata']
-            entity = ImageRegistry(self, meta['name'],
-                                   entity_j['status']['dockerImageRepository'],
-                                   meta['namespace'])
-            if entity not in entities:
-                entities.append(entity)
-        return entities
+        namespace = namespace if namespace else self.default_namespace
```

Comment:
If the user does not specify a namespace, why lock it in to default? They should specify default if they want routes from the default namespace